lib/datapacksutils.js

```diff
@@ -4917,4 +4917,52 @@ var SALESFORCE_FOLDER_TO_TYPE =  {
     triggers: "ApexTrigger",
     weblinks: "CustomPageWebLink",
     workflows: "Workflow"
-}	
\ No newline at end of file
+}
+
+DataPacksUtils.prototype.updateDataPackWithNewMappings = function (dataPack, jobInfo) {
+    let obj = {};
+    let dataPackType = dataPack.VlocityDataPackType;
+
+    let dataField = this.getDataField(dataPack);
+    let mappings = jobInfo.platformObjectMappings[dataPackType][dataField];
+
+    if (mappings) {
+        let newDataField = mappings.objectName;
+        obj[newDataField] = dataPack.VlocityDataPackData[dataField];
+
+        //Iterate through keys of <dataPack.VlocityDataPackData[dataField]> to update the keys(fieldNames)
+        Object.keys(obj[newDataField][0]).forEach(key => {
+            if (Object.keys(mappings.fieldNames).includes(key) && mappings.fieldNames[key] !== key) {
+                obj[newDataField][0][mappings.fieldNames[key]] = obj[newDataField][0][key];
+                delete obj[newDataField][0][key];
+            }
+            if (Object.keys(jobInfo.platformObjectMappings[dataPackType]).includes(key)) {
+                let subMappings = jobInfo.platformObjectMappings[dataPackType][key];
+                let childObjDataField = subMappings.objectName;
+                obj[newDataField][0][childObjDataField] = [];
+
+                // Iterate through array of Level 2 Objects
+                // Level 2 object like DRMapItems__c in DRBundle__c & Element__c in OmniScript__c
+                obj[newDataField][0][key].forEach(subObj => {
+                    let level2Obj = subObj;
+                    Object.keys(subObj).forEach(fieldName => {
+                        if (subMappings.fieldNames[fieldName]) {
+                            level2Obj[subMappings.fieldNames[fieldName]] = subObj[fieldName];
+                            if (fieldName !== subMappings.fieldNames[fieldName]) {
+                                delete level2Obj[fieldName];
+                            }
+                        }
+                    });
+                    level2Obj.VlocityRecordSObjectType = childObjDataField;
+                    level2Obj.VlocityRecordSourceKey = level2Obj.VlocityRecordSourceKey.replace(subObj.VlocityRecordSObjectType, childObjDataField);
```

Comment:
This replace will only replace the frst instance in JS. Needs to use the replace all syntax. `res.replace(new RegExp(subObj.VlocityRecordSObjectType, 'g'), childObjDataField)`